src/game.cpp

```diff
@@ -4078,9 +4078,27 @@ bool Game::combatChangeHealth(Creature* attacker, Creature* target, CombatDamage
 		message.position = targetPos;
 
 		SpectatorVec spectators;
-		if (targetPlayer && target->hasCondition(CONDITION_MANASHIELD) && damage.primary.type != COMBAT_UNDEFINEDDAMAGE) {
+		if (targetPlayer && (target->hasCondition(CONDITION_MANASHIELD) || target->hasCondition(CONDITION_MANASHIELD_BREAKABLE)) && damage.primary.type != COMBAT_UNDEFINEDDAMAGE) {
 			int32_t manaDamage = std::min<int32_t>(targetPlayer->getMana(), healthChange);
-			if (manaDamage != 0) {
+			if (g_config.getBoolean(ConfigManager::MANASHIELD_BREAKABLE) && targetPlayer) {
+				// Mana Shield Breakable
+				uint16_t manaShield = targetPlayer->getManaShield();
+				if (manaShield > 0) {
+					if (manaShield > manaDamage) {
+						targetPlayer->setManaShield(manaShield - manaDamage);
+						manaShield = manaShield - manaDamage;
+					} else {
+						manaDamage = manaShield;
+						targetPlayer->removeCondition(CONDITION_MANASHIELD_BREAKABLE);
+						manaShield = 0;
+					}
+				}
+				if (targetPlayer->getMana() == 0 && manaShield > 0) {
+					targetPlayer->removeCondition(CONDITION_MANASHIELD_BREAKABLE);
+				}
+			} else {
+			// Old Mana Shield
```

Comment:
normal mana shield?

Reply:
old system that use all mana and cannot be broken when mana reach value 0

Reply:
old system that use all mana and cannot be broken when mana reach value 0

Reply:
I think nekiro means to rename "old mana shield" to "normal mana shield"